hack/README.md

```diff
@@ -0,0 +1,55 @@
+# Karmada hack GuideLines
+
+This document describes how you can use the scripts from the [`hack`](.) directory
+and gives a brief introduction and explanation of these scripts.
+
+Scripts in this directory are mainly for the purpose which improves development efficiency and
+ensures development quality
+
+# Key scripts
+
+## For end-user
+
+- [`local-up-karmada.sh`](local-up-karmada.sh) This script will quickly pull up a local Karmada environment, that contains
+  a host K8s cluster by [kind](https://kind.sigs.k8s.io/ "A tool for running local Kubernetes clusters using Docker container"),
+  and will compile all components local image to startup in this cluster.
+  
+- [`remote-up-karmada.sh`](remote-up-karmada.sh) This script will install Karmada to a standalone K8s cluster, this cluster
+  may be real, remote , and even for production. It is worth noting for the connectivity from your client to Karmada API server,
+  it will create a load balancer service with an external IP by default, if your want to customize this service, you may add 
+  the annotations at the metadata part of service `karmada-apiserver` in 
+  [`../artifacts/deploy/karmada-apiserver.yaml`](../artifacts/deploy/karmada-apiserver.yaml) before the installing. The 
+  following is an example.
+```
+  # If you want to use a internal IP in public cloud you need to fill the following annotation, 
+  # Fot the more annotation settings please read your public cloud docs
+  annotations: 
+    # Aliyun cloud
+    #service.beta.kubernetes.io/alibaba-cloud-loadbalancer-address-type: "intranet"
+    # Huawei cloud
+    #kubernetes.io/elb.autocreate: '{"type":"inner"}'
+    # Tencent cloud (you need to replace words 'xxxxxxxx')
+    #service.kubernetes.io/qcloud-loadbalancer-internal-subnetid: subnet-xxxxxxxx
+```
+
+- [`deploy-karmada-agent.sh`](deploy-karmada-agent.sh) This script will install Karmada Agent to your member cluster.
+
+- [`undeploy-karmada.sh`](undeploy-karmada.sh) This script will uninstall Karmada from your local environment, it will 
+  remove Karmada control plane from the host cluster, and will not remove your Docker container by kind. If you 
```

Comment:
maybe I should say "remove Karmada control plane from the host cluster without removing Docker container created by kind"
I mean that the `undeploy-karmada.sh` script will not remove the `karmada-host` cluster, actually, it's a Docker container created by kind

Reply:
`This script will uninstall Karmada from the specific cluster.` 

How about this? Maybe we don't need to explain if it will remove the cluster.